picard/ui/options/maintenance.py

```diff
@@ -0,0 +1,166 @@
+        current_options = set([
+            # Include options that are required but are not entered directly from the options pages.
+            'file_renaming_scripts',
+            'selected_file_naming_script_id',
+            'log_verbosity',
+            # Items missed if TagsCompatibilityWaveOptionsPage does not register.
+            'remove_wave_riff_info',
+            'wave_riff_info_encoding',
+            'write_wave_riff_info',
+        ]).union(key_options)
+
+        # All setting options included in the INI file.
+        config.beginGroup("setting")
+        file_options = set(config.childKeys())
+        config.endGroup()
+
+        orphan_options = file_options.difference(current_options)
+
+        self.ui.description.setText(
+            _(self.INSTRUCTIONS)
+            + _("\n\nThe configuration file currently contains {0} option settings, {1} which are unused.").format(
+                len(file_options),
+                len(orphan_options)
+            )
+        )
+        self.ui.enable_cleanup.setChecked(False)
+        self.ui.tableWidget.clearContents()
+        self.ui.tableWidget.setRowCount(len(orphan_options))
+        for row, option_name in enumerate(sorted(orphan_options)):
+            tableitem = QtWidgets.QTableWidgetItem(option_name)
+            tableitem.setData(QtCore.Qt.UserRole, option_name)
+            tableitem.setFlags(tableitem.flags() | QtCore.Qt.ItemIsUserCheckable)
+            tableitem.setCheckState(False)
+            self.ui.tableWidget.setItem(row, 0, tableitem)
+            tableitem = QtWidgets.QTextEdit()
+            text = self.make_setting_value_text(option_name)
+            tableitem.setPlainText(text)
+            tableitem.setReadOnly(True)
+            # Adjust row height to reasonably accommodate values with more than one line, with a minimum
+            # height of 25 pixels.  Multi-line values will be expanded to display up to 5 lines, assuming
+            # a standard line height of 18 pixels.
+            row_height = max(25, 18 * min(5, text.count("\n") + 1))
+            self.ui.tableWidget.setRowHeight(row, row_height)
+            self.ui.tableWidget.setCellWidget(row, 1, tableitem)
+        self.ui.tableWidget.resizeColumnsToContents()
+        self.ui.select_all.setCheckState(False)
+        self.enable_cleanup_changed()
+
+    def save(self):
+        if not self.ui.enable_cleanup.checkState() == QtCore.Qt.Checked:
+            return
+        to_remove = set()
+        for idx in range(self.ui.tableWidget.rowCount()):
+            item = self.ui.tableWidget.item(idx, 0)
+            if item.checkState() == QtCore.Qt.Checked:
+                to_remove.add(item.data(QtCore.Qt.UserRole))
+        if to_remove and QtWidgets.QMessageBox.question(
+            self,
+            _('Confirm Remove'),
+            _("Are you sure you want to remove the {0} selected option settings?").format(len(to_remove)),
```

Comment:
Perhaps we should keep it simple and just remove number of settings to remove in this message, else we'll have to handle plural form (feasible of course, but is it worth?):

![image](https://user-images.githubusercontent.com/151042/131881274-0fdcdf00-01a2-4898-9dc0-cf02fbc8df92.png)

What about simplifying to:

```python
_("Are you sure you want to remove selected option settings?"),
```

After all, there's one checkbox to enable the deletion, then one for each setting, plus a confirmation message; the number of selected settings isn't really needed imho.

If not, then we need to use `ngettext()` instead of `_()`.

Reply:
Agreed that the number can be removed.  I actually had this in for testing during development and forgot to remove it.